io/io/test/TFileMergerTests.cxx

```diff
@@ -0,0 +1,48 @@
+#include "TFileMerger.h"
+
+#include "TMemFile.h"
+#include "TTree.h"
+
+#include "gtest/gtest.h"
+
+static void CreateATuple(TMemFile &file, const char *name, double value)
+{
+   auto mytree = new TTree(name, "A tree");
+   mytree->Branch(name, &value);
+   mytree->Fill();
+   file.Write();
+}
+
+static void CheckTree(TMemFile &file, const char *name, double expectedValue)
+{
+   auto t = static_cast<TTree *>(file.Get(name));
+   ASSERT_TRUE(t != nullptr);
+
+   double d;
+   t->SetBranchAddress(name, &d);
```

Comment:
Setting branch address to a stack address requires to either call ```t->ResetBranchAddresses();``` or delete the TTree (otherwise it is technically in an invalid state).

Reply:
Does that mean that `file.Get(name)` allocates an object and it is the user's responsibility to deallocate it? If yes, I assume I can transform it into a unique_ptr and would that work well with having a `t->ResetBranchAddresses();`?

Reply:
> Does that mean that file.Get(name) allocates an object and it is the user's responsibility to deallocate it?

In the case of a TTree, the point is an implicit shared pointer.  You can delete it (and the TFile will be informed) and you can let the TFile delete it (but your are not directly inform when it does unless you are registered in the list of cleanups).

> If yes, I assume I can transform it into a unique_ptr 

Yes, you can as long as you make sure that the lifetime of the unique_ptr is less or equal to the litetime of the TFile.

Reply:
Should be done.